### PR TITLE
BOT API 7.3

### PR DESCRIPTION
**Pull request by Badiboy**
Bot API 7.3

- [x] _Added support for [InlineKeyboardMarkup](https://core.telegram.org/bots/api#inlinekeyboardmarkup) with url, login_url, and callback_game buttons for messages sent on behalf of a business account._
- [x] Added the field via_join_request to the class [ChatMemberUpdated](https://core.telegram.org/bots/api#chatmemberupdated).
- [x] Added support for live locations that can be edited indefinitely, allowing 0x7FFFFFFF to be used as live_period.
- [x] Added the parameter live_period to the method [editMessageLiveLocation](https://core.telegram.org/bots/api#editmessagelivelocation).
- [x] Added the field question_entities to the class [Poll](https://core.telegram.org/bots/api#poll).
- [x] Added the field text_entities to the class [PollOption](https://core.telegram.org/bots/api#polloption).
- [x] Added the parameters question_parse_mode and question_entities to the method [sendPoll](https://core.telegram.org/bots/api#sendpoll).
- [x] Added the class [InputPollOption](https://core.telegram.org/bots/api#inputpolloption) and changed the type of the parameter options in the method [sendPoll](https://core.telegram.org/bots/api#sendpoll) to Array of [InputPollOption](https://core.telegram.org/bots/api#inputpolloption).
- [x] Added the classes [ChatBackground](https://core.telegram.org/bots/api#chatbackground), [BackgroundType](https://core.telegram.org/bots/api#backgroundtype), [BackgroundFill](https://core.telegram.org/bots/api#backgroundfill) and the field chat_background_set of type [ChatBackground](https://core.telegram.org/bots/api#chatbackground) to the class [Message](https://core.telegram.org/bots/api#message), describing service messages about background changes.
- [x] Split out the class [ChatFullInfo](https://core.telegram.org/bots/api#chatfullinfo) from the class [Chat](https://core.telegram.org/bots/api#chat) and changed the return type of the method [getChat](https://core.telegram.org/bots/api#getchat) to [ChatFullInfo](https://core.telegram.org/bots/api#chatfullinfo).
- [x] Added the field max_reaction_count to the class [ChatFullInfo](https://core.telegram.org/bots/api#chatfullinfo).
- [x] Documented that .MP3 and .M4A files can be used as voice messages.

### CODE REVIEW

**Comment by Badiboy** [comment]
@coder2020official I'm done.

Regarding this:

> Split out the class [ChatFullInfo](https://core.telegram.org/bots/api#chatfullinfo) from the class [Chat](https://core.telegram.org/bots/api#chat) and changed the return type of the method [getChat](https://core.telegram.org/bots/api#getchat) to [ChatFullInfo](https://core.telegram.org/bots/api#chatfullinfo).

https://github.com/eternnoir/pyTelegramBotAPI/pull/2264/commits/3532812e0ed6d44675a4f16c4e2efd3c57b3dd1d

I currently left "Chat" as a full copy of "ChatFullInfo". It will left compatibility without making tenths of stubs with warnings. Later we can do it if anyone will have time for that.

**Comment by coder2020official** [comment]
Got your message.
I will try reviewing tonight

**Review by coder2020official — APPROVED** [review]
Otherwise, it looks good to me...

File: telebot/types.py [hunk 1]
```diff
@@ -6972,6 +6975,7 @@ def __init__(
         self.correct_option_id: int = correct_option_id
         self.explanation: str = explanation
         self.explanation_entities: List[MessageEntity] = explanation_entities
+        self.question_entities: List[MessageEntity] = question_entities
```
[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/types.py:7032]: I did it in the same way as other entities in this class. Thus if we want to have some function (I did not get which one) - we can do it later for all such entities.

File: telebot/apihelper.py [hunk 2]
```diff
@@ -439,6 +439,8 @@ def copy_message(token, chat_id, from_chat_id, message_id, caption=None, parse_m
         payload['disable_notification'] = disable_notification
     if reply_parameters is not None:
         payload['reply_parameters'] = reply_parameters.to_json()
+    if reply_markup is not None:
+        payload['reply_markup'] = await _convert_markup(reply_markup)
```
[the remark this answers is not in the document]
Comment (Badiboy) [reply telebot/apihelper.py]: It was missed in sync. Async was fine.

File: telebot/types.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (coder2020official) [reply telebot/types.py:7032]: something like html_caption or html_text

**Pull request merged**